source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/VideoViewResizeable.java

```diff
@@ -0,0 +1,87 @@
+package io.adaptivecards.renderer.readonly;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.widget.VideoView;
+
+import java.util.HashMap;
+
+// Taken from here: https://stackoverflow.com/questions/13603553/videoview-to-match-parent-height-and-keep-aspect-ratio/40475998
+
+public class VideoViewResizeable extends VideoView
+{
+    public VideoViewResizeable(Context context)
+    {
+        super(context);
+    }
+
+    public void retrieveVideoDimensions(String uri, boolean isOnlineSource)
+    {
+        MediaMetadataRetriever metadataRetriever = new MediaMetadataRetriever();
+
+        if(isOnlineSource)
+        {
+            metadataRetriever.setDataSource(uri, new HashMap<String, String>());
+        }
+        else
+        {
+            metadataRetriever.setDataSource(getContext(), Uri.parse(uri));
+        }
+
+        m_videoHeight = Integer.parseInt(metadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT));
+        m_videoWidth = Integer.parseInt(metadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH));
+    }
+
+    public void setVideoPath(String path, boolean isAudio)
+    {
+        m_isAudio = isAudio;
+        if(!m_isAudio)
+        {
+            retrieveVideoDimensions(path, false);
+        }
+        super.setVideoPath(path);
+    }
+
+    public void setVideoURI(String uri, boolean isAudio)
+    {
+        m_isAudio = isAudio;
+        if(!m_isAudio)
+        {
+            retrieveVideoDimensions(uri, true);
+        }
+        super.setVideoURI(Uri.parse(uri));
+    }
+
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec)
+    {
+        int width = getDefaultSize(m_videoWidth, widthMeasureSpec);
+        int height = getDefaultSize(m_videoHeight, heightMeasureSpec);
+        if (m_videoWidth > 0 && m_videoHeight > 0)
+        {
+            if ((m_videoWidth * height) > (width * m_videoHeight))
+            {
+                height = (width * m_videoHeight) / m_videoWidth;
+            }
+            else if ((m_videoWidth * height) < (width * m_videoHeight))
+            {
+                width = (height * m_videoWidth) / m_videoHeight;
+            }
+        }
+        else
+        {
+            // Give some height so the media controller fits
+            height = 200;
+        }
```

Comment:
Should there be a minimum height?

Reply:
when the media is audio we have to provide a minimum size, otherwise the media player dissapears

---
In reply to: [208019758](https://github.com/Microsoft/AdaptiveCards/pull/1716#discussion_r208019758) [](ancestors = 208019758)